packages/scripts/create-version.sh

```diff
@@ -1,86 +1,137 @@
-      log "Error during fetching data from repository"
-      false
-    }
-    git pull || {
-      log "Error during pulling ${MAIN_BRANCH} branch"
-      false
-    }
-    git diff origin/${MAIN_BRANCH} --exit-code --compact-summary || {
-      log "Working directory not clean. Delete or push your changes to ${MAIN_BRANCH}}."
-      false
-    }
+error() {
+    log "ERROR: " $@
+}
 
-    log "Running tests and build..."
-    npm run test || {
-      log "Running tests failed"
-      false
-    }
-    npm run build || {
-      log "Running build failed"
-      false
-    }
+VERSION_TYPE=$1
+PACKAGE_NAME=$2
 
-    NEXT_VERSION=$(npm version --no-git-tag-version "$@")
-    log "Creating ${NPM_PACKAGE_NAME} ${NEXT_VERSION}...";
+if [ -z "$VERSION_TYPE" ]; then
+    error "No version type provided."
+    help
+fi
 
-    log "Creating version publish branch...";
-    NEXT_VERSION_BRANCH="publish-${NEXT_VERSION}"
-    git checkout -b ${NEXT_VERSION_BRANCH}
-    git add package.json package-lock.json
-    git commit -m "Bump version to ${NEXT_VERSION}"
-    git tag ${NEXT_VERSION}
+if [ -n "$PACKAGE_NAME" ]; then
+    IS_WORKSPACE_PACKAGE=1
+    PACKAGE_DIRECTORY="packages/$PACKAGE_NAME"
+    cd $PACKAGE_DIRECTORY || {
+        error "Invalid package name provided. Directory '$PACKAGE_DIRECTORY' does not exist."
+        false
+    }
+fi
+NPM_PACKAGE_NAME=`cut -d "=" -f 2 <<< $(npm run env | grep "npm_package_name")`
 
-    log "This is the last step before CI-based npm publish.";
-    log "Are you sure you want to trigger ${NPM_PACKAGE_NAME} ${NEXT_VERSION} version publish?";
-    select choice in "Yes" "No"; do
-    case "$choice" in
-        Yes)
-            log "Pushing changes...";
-            git push origin ${NEXT_VERSION_BRANCH} tag ${NEXT_VERSION}
-            git checkout ${MAIN_BRANCH}
-            break;;
-        *)
-            log "Reverting changes...";
-            git checkout ${MAIN_BRANCH}
-            git tag -d ${NEXT_VERSION}
-            git branch -D ${NEXT_VERSION_BRANCH}
-            break;;
-    esac
-    done
+MAIN_BRANCH=${MAIN_BRANCH:-master}
+log "Switching to main branch (${MAIN_BRANCH}) and checking if its up-to-date...";
+git checkout ${MAIN_BRANCH} || {
+  error "Error during checking out ${MAIN_BRANCH} branch"
+  false
+}
+git fetch || {
+  error "Error during fetching data from repository"
+  false
 }
+git pull || {
+  error "Error during pulling ${MAIN_BRANCH} branch"
+  false
+}
+git diff origin/${MAIN_BRANCH} --exit-code --compact-summary || {
+  error "Working directory not clean. Delete or push your changes to ${MAIN_BRANCH}}."
+  false
+}
+
+log "Running tests and build..."
+npm run lint --if-present || {
+  error "Running lint check failed"
+  false
+}
+npm run check-types --if-present || {
+  error "Running types check failed"
+  false
+}
+npm run test --if-present || {
+  error "Running tests failed"
+  false
+}
+npm run build --if-present || {
+  error "Running build failed"
+  false
+}
```

Comment:
We have `set -e` at the beginning of the script. So what's the purpose of these conditional constructs?

Reply:
I think it's here to add user-friendly error log.

Reply:
Personally I think it helps in some cases, so I would leave it.

Example:
```
Your branch is up to date with 'origin/RD-5955'.
Already up to date.
 packages/backend/test/logger.test.ts | 2 +-
 packages/scripts/create-version.sh   | 6 ++----
 2 files changed, 3 insertions(+), 5 deletions(-)
create-version.sh: ERROR: Working directory not clean. Delete or push your changes to RD-5955.
```

IMHO without "Working directory not clean. Delete or push your changes to RD-5955." it wouldn't be clear what to do next.

I'm just adding small fix - https://github.com/cloudify-cosmo/cloudify-ui-common/pull/139/commits/ff9823cc22a210989bed8eab41160ccd3b1a7ff9 .